kv/kvStore.go

```diff
@@ -5,42 +5,58 @@ import (
 	"fmt"
 )
 
-const MaxMem = 1 << (10 * 3) // 1 GB
-const defPath = "."          // create in local directory
+const MaxMem = 1 << (10 * 3) // Do not allow KV stores to use more than 1GB of memory
+const DefaultPath = "."      // Default to current working directory to persist KV store
 
+// KeyValueStore defines the interface to be implemented by the KV store.
 type KeyValueStore interface {
-	Put(key uint64, value [10]byte)
-	Get(key uint64) [10]byte
-	Create(config KvStoreConfig)
-	Open(path string)
-	Delete(path string)
-	Close()
+	// Put stores a new item with given key and value in the KV store. If
+	// an item with the requested key already exists, an error is returned.
+	Put(key uint64, value [10]byte) error
+	// Get retrieves an item with given key rom the KV store. If no item
+	// with the requested key exists, an error is returned.
+	Get(key uint64) ([10]byte, error)
+	// Create initializes a new instance of the KV store with the supplied
+	// parameters. If creation fails, an error is returned.
+	Create(config KvStoreConfig) error
+	// Open opens an existing KV store from disk. If loading fails, an
+	// error is returned.
+	Open(path string) error
+	// Delete deletes the currently opened KV store. If deletion fails, an
+	// error is returned.
+	Delete() error
+	// Close persists the active KV store to disk and unloads it. If it
+	// fails, an error is returned.
+	Close() error
 }
 
+// KVStoreConfig provides parameters used to initialize a new KV store.
 type KvStoreConfig struct {
-	memorySize       int
-	workingDirectory string
+	memorySize       int    // Maximum amount of memory to be used by KV store
```

Comment:
Wouldn't it make sense to make `size` of type `uint`?
It would automatically deny negative values.